lib/elixir/lib/module/types/helpers.ex

```diff
@@ -203,12 +218,34 @@ defmodule Module.Types.Helpers do
   @doc """
   Converts the given expression to a string,
   translating inlined Erlang calls back to Elixir.
+
+  We also undo some macro expresions done by the Kernel module.
   """
   def expr_to_string(expr) do
     expr
     |> Macro.prewalk(fn
-      {{:., _, [mod, fun]}, meta, args} -> erl_to_ex(mod, fun, args, meta)
-      other -> other
+      {:%, _, [Range, {:%{}, _, fields}]} = node ->
+        case :lists.usort(fields) do
```

Comment:
Would `maps:from_list/1` here make this code easier to read, so that we don't need to do the mental math to check that the fields below are ordered? 😬